### PR TITLE
Bug 1188716 - fix: add a defer to block user using the slider

### PR DESCRIPTION
**Pull request by snowmantw**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=776c027e25d553e2bbaf08df8f35477251ea9907

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ed424195d3c14bf7d0d48b9c74d00645df51166b

File: apps/system/lockscreen/js/lockscreen.js [hunk 1]
```diff
@@ -518,10 +530,17 @@
...
+      value = val == 'false' ? false : true;
     } else {
-      this.passCodeEnabled = val;
+      value = val;
+    }
+
+    if (this.passCodeEnabled instanceof LockScreen.Deferred) {
+      this.passCodeEnabled.resolve(value);
+    } else {
+      this.passCodeEnabled = value;
```
Comment (timdream) [inline apps/system/lockscreen/js/lockscreen.js]: This line should be moved out of the `else` block right to move `this.passCodeEnabled` back into a boolean?

File: apps/system/lockscreen/js/lockscreen.js [hunk 2]
```diff
@@ -336,6 +336,18 @@
     window.addEventListener('emergency-call-leave', this);
   };
 
+  LockScreen.prototype.setupPassCodeEnabled =
+  function() {
+    this.passCodeEnabled = new LockScreen.Deferred();
+    return this.passCodeEnabled.promise.then((val) => {
+      // Replace the pending request.
+      this.passCodeEnabled = val;
```
Comment (timdream) [inline apps/system/lockscreen/js/lockscreen.js]: Never mind, I see this.

**Pull request merged**